kml.cc

```diff
@@ -288,13 +290,21 @@ const char* kml_tags_to_ignore[] = {
   NULL,
 };
 
+// The TimeSpan/begin and TimeSpan/end DateTimes:
+static gpsbabel::DateTime wpt_timespan_begin, wpt_timespan_end;
+
 void wpt_s(xg_string, const QXmlStreamAttributes*)
 {
   if (wpt_tmp) {
     fatal(MYNAME ": wpt_s: invalid kml file\n");
   }
   wpt_tmp = new Waypoint;
   wpt_tmp_queued = 0;
+
+  /* Invalidate timespan elements for a beginning Placemark,
+   * so that each Placemark has its own (or no) TimeSpan. */
+  wpt_timespan_begin.setTime_t(0);
+  wpt_timespan_end.setTime_t(0);
```

Comment:
Can you use the gpsbabel::DateTime() constructor here instead of setTime_t(0)?  This encapsulates the workings of gpsbabel::DateTime better, and will avoid trouble in the unlikely event that the crutch is removed.
`  // As a crutch, mimic the old behaviour of an uninitialized creation time
  // being 1/1/1970.
`
For example:
gtrnctr.cc:  gtc_least_time = gpsbabel::DateTime();
gtrnctr.cc:  gtc_most_time = gpsbabel::DateTime();